docs/troubleshoot-collection.md

```diff
@@ -239,6 +248,63 @@ You [check Prometheus logs](#prometheus-logs) to verify there are no errors duri
 
+### Check Prometheus memory usage
+
+Verify memory usage with `kubectl`. First get the actual usage with:
+
+```
+kubectl top pod -n <DEP>-collections -l app.kubernetes.io/name=prometheus
+NAME                                                   CPU(cores)   MEMORY(bytes)
+prometheus-stag-collection-prometheus-0-state-0        89m          2853Mi
+prometheus-stag-collection-prometheus-1-controller-0   146m         2991Mi
+prometheus-stag-collection-prometheus-2-kubelet-0      82m          2845Mi
+prometheus-stag-collection-prometheus-3-container-0    131m         3045Mi
+prometheus-stag-collection-prometheus-4-container-0    102m         2817Mi
+prometheus-stag-collection-prometheus-5-node-0         133m         2989Mi
+prometheus-stag-collection-prometheus-6-operator-r-0   182m         2968Mi
+prometheus-stag-collection-prometheus-7-prometheus-0   114m         3283Mi
+prometheus-stag-collection-prometheus-8-control-pl-0   121m         3017Mi
+prometheus-stag-collection-prometheus-9-sumo-0         155m         2953Mi
+prometheus-stag-collection-prometheus-assembly-0       50m          821Mi
+```
+
+Now get the memory limits of prometheus containers from the pods:
+
+```
+kubectl get pod -n stag-collections -l app.kubernetes.io/name=prometheus -o "custom-columns=NAME:.metadata.name,CONTAINERS:.spec.containers[*].name,MEM_LIMIT:.spec.containers[*].resources.limits.memory"
+NAME                                                   CONTAINERS                   MEM_LIMIT
+prometheus-stag-collection-prometheus-0-state-0        prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-1-controller-0   prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-2-kubelet-0      prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-3-container-0    prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-4-container-0    prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-5-node-0         prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-6-operator-r-0   prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-7-prometheus-0   prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-8-control-pl-0   prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-9-sumo-0         prometheus,config-reloader   20Gi,50Mi
+prometheus-stag-collection-prometheus-assembly-0       prometheus,config-reloader   20Gi,50Mi
+```
+
+Note that pods can contain more than one container so in this example prometheus pods have 20Gi memory limit and config-reloader containers
+have 50Mi memory limit.
+
+When memory usage is higher than ~`95%` of Prometheus container's memory limit, (in the above case it's not: none of the containers exceeded
+`95% * 20 = 19Gi` of used memory) remove WAL from within the container:
```

Comment:
I don't think this is proper solution. I would rather say that customer should increase memory requests/limits for Prometheus or reduce number of metrics being scraped by the Prometheus. Removing WAL is more like temporary solution

Reply:
yes, especially because it is temporary solution or clean up after/during metrics spike